src/validation-engine.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  ValidationResult,
+  FormValidationResult,
+  FieldsValidationSchema,
+  RecordValidationFunction,
+  createDefaultValidationResult,
+  RecordValidationFunctionSyncAsync,
+  isSyncValidationResult,
+  FieldValidationFunctionSyncAsync,
+  FieldValidationFunction,
+} from './model';
+
+import { isUndefinedOrNull } from './helper';
+
+import {
+  fireAllFieldsValidations,
+  fireSingleFieldValidations,
+  fireRecordValidations,
+} from './validation-dispatcher';
+import { buildFormValidationResult } from './form-validation-summary-builder';
+
+export class ValidationEngine {
```

Comment:
Do we still need the class?

Reply:
I was making my self the same question, the thing is we've got three member variables that need to be unique by instance: validationsPerFeild, validationGlobalform and asyncValidationInProgressCount.

ValidationEngine is created once per form, and used in the formValidationBase whenever it's needed.

How could we make the same using functions? Not sure if it's a good idea to use something like revealing module pattern but smells to ES5.

Any other approach?

Reply:
In `lcFormValidation` there is `baseFormValidation.ts` file that it contains the class `BaseFormValidation`.
This is the real instance that users need to create on each form. Underhood, we could keep ValidationEngine as functions but providing the necessary params on each methods.

What do you think?

Reply:
Depends on the number of parameters we need to pass, if we have one or two is ok, but more than 3 could be a mess

Reply:
Should I jump implementing baseFormValidation then check if there are no many parameters involved and refactor? We could even create a structure to simplify this, not sure if it would be a good idea because validationEngine will mutate params